pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -654,6 +654,7 @@ func (aws *AWSCloud) NodeAddresses(name string) ([]api.NodeAddress, error) {
 		}
 		return []api.NodeAddress{
 			{Type: api.NodeInternalIP, Address: internalIP},
+			{Type: api.NodeLegacyHostIP, Address: internalIP},
```

Comment:
I would rather not do this, because this then is returned to clients via the API, and we are trying to remove NodeLegacyHostIP.  How hard would it be to change the tests?